src/App.tsx

```diff
@@ -15,48 +29,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function getPreparedGoods(
+  goods: Goods,
+  { sortField, isReversed }: FilterParams,
+) {
+  let allGoods;
+
+  switch (sortField) {
+    case SortType.Alphabet:
+      allGoods = [...goods].sort();
+      break;
+    case SortType.Length:
+      allGoods = [...goods].sort((good1, good2) => good1.length - good2.length);
+      break;
+    case SortType.NoSort:
+    default:
+      allGoods = [...goods];
+      break;
+  }
+
+  if (isReversed) {
+    allGoods.reverse();
+  }
+
+  return allGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  function handleSort(field: SortType) {
+    return () => setSortField(field);
+  }
+
+  const handleReverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const handleReset = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortType.Alphabet,
+          })}
+          onClick={handleSort(SortType.Alphabet)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SortType.Length,
+          })}
+          onClick={handleSort(SortType.Length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={handleReverse}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {JSON.stringify(visibleGoods) !== JSON.stringify(goodsFromServer) && (
+          <button
```

Comment:
better use variable, for example `showResetButton` with conditions by sortField and isReversed than this parsing

Reply:
agree with [comment](https://github.com/mate-academy/react_list-of-goods-hooks/pull/1082/files#r1253992108)